schema/bom-1.3-SNAPSHOT.proto

```diff
@@ -0,0 +1,391 @@
+syntax = "proto3";
+package org.cyclonedx.schema.bom._1_3;
+
```

Comment:
One thing to consider in general for protobuf is `optional` https://stackoverflow.com/questions/42622015/how-to-define-an-optional-field-in-protobuf-3/62566052#62566052

There are some nice advantages to doing this for an upstream consumer, because code gen can create methods to check if a field exists, etc...

If a field is truly optional, this would be nice for people consuming. It will prevent a lot of null checks, etc... I imagine.

Reply:
@DarthHater Do you know how `optional` works on the wire and in practice?

I was just reading the documentation here https://developers.google.com/protocol-buffers/docs/proto#optional. I don't really understand the value unless you want to assign some kind of non-default default value.

But, then reading this https://github.com/protocolbuffers/protobuf/blob/v3.12.0/docs/field_presence.md and it seems to do more.

Is it actually storing an extra information as to the presence of a null value? Or just indicating to codegen that it should create helper methods?

Reply:
Ah, sorry, that first link is to the proto2 documentation.